src/components/radio/radio_button.ts

```diff
@@ -0,0 +1,254 @@
+import {
+  AfterContentInit,
+  Attribute,
+  Component,
+  ContentChildren,
+  Directive,
+  HostBinding,
+  HostListener,
+  Input,
+  Optional,
+  Output,
+  QueryList,
+  ViewEncapsulation,
+  forwardRef
+} from 'angular2/core';
+
+import {ObservableWrapper, EventEmitter} from 'angular2/src/facade/async';
+import {iterateListLike} from 'angular2/src/facade/collection';
+import {Event} from 'angular2/src/facade/browser';
+
+import {MdRadioDispatcher} from './radio_dispatcher';
+
+// TODO(mtlin):
+// Ink ripple is currently placeholder.
+// Determine motion spec for button transitions.
+// Design review.
+// RTL
+// Support forms API.
+// Use ChangeDetectionStrategy.OnPush
+// Flesh out MdRadioChange.
+
+var _uniqueIdCounter = 0;
+
+export class MdRadioChange {}
+
+@Directive({
+  selector: 'md-radio-group',
+  host: {
+    'role': 'radiogroup',
+  },
+})
+export class MdRadioGroup implements AfterContentInit {
+  /** The value for the radio group. Should match currently selected button. */
+  @Input('value')
+  private _value: MdRadioChange = null;
+
+  /** The HTML name attribute applied to radio buttons in this group. */
+  @Input('name')
+  private _name: string = null;
+
+  /** Disables all individual radio buttons assigned to this group. */
+  @Input('disabled')
+  private _disabled: boolean = false;
+
+  /** The currently selected radio button. Should match value. */
+  @Input('selected')
+  private _selected: MdRadioButton = null;
+
+  /** Event emitted when the group value changes. */
+  @Output() change: EventEmitter<any> = new EventEmitter();
+
+  /** Child radio buttons. */
+  @ContentChildren(forwardRef(() => MdRadioButton))
+  private _radios: QueryList<MdRadioButton> = null;
+
+  /**
+   * Initialize properties once content children are available.
+   * This allows us to propagate relevant attributes to associated buttons.
+   */
+  ngAfterContentInit() {
+    if (this._name == null) {
+      this.name = `md-radio-group-${_uniqueIdCounter++}`;
+    }
+    // Trigger `name` attribute propagation.
+    this.name = this.name;
```

Comment:
I don't follow why this is necessary. If you're relying on this forcing the setter to be called, there should instead be a separate function that's called both here and in the setter.